app/services/rapidfire/answer_group_builder.rb

```diff
@@ -19,10 +19,10 @@ def save!(options = {})
           # in case of checkboxes, values are submitted as an array of
           # strings. we will store answers as one big string separated
           # by delimiter.
+          text = text.values if text.is_a?(Hash)
```

Comment:
@iffyuva as you can see in `_checkbox` template changes, I changed `[]` name convention to `[1]`, `[2]` ... This is used to give checkbox ids a proper `id` attribute. Since this, server will get chosen checkbox values as a Hash, not an Array.